commands/src/main/java/org/wildfly/extras/creaper/commands/elytron/sasl/AddConfigurableSaslServerFactory.java

```diff
@@ -0,0 +1,226 @@
+            if (filters == null) {
+                throw new IllegalArgumentException("Filters added to configurable-sasl-server-factory must not be null");
+            }
+            Collections.addAll(this.filters, filters);
+            return this;
+        }
+
+        public Builder addProperties(Property... properties) {
+            if (properties == null) {
+                throw new IllegalArgumentException("Properties added to configurable-sasl-server-factory must not be null");
+            }
+            Collections.addAll(this.properties, properties);
+            return this;
+        }
+
+        public Builder replaceExisting() {
+            this.replaceExisting = true;
+            return this;
+        }
+
+        public AddConfigurableSaslServerFactory build() {
+            if (saslServerFactory == null || saslServerFactory.isEmpty()) {
+                throw new IllegalArgumentException("sasl-server-factory must not be null and must include at least one entry");
+            }
+            return new AddConfigurableSaslServerFactory(this);
+        }
+
+    }
+
+    public static final class Filter {
+
+        private final String patternFilter;
+        private final String predefinedFilter;
+        private final Boolean enabling;
+
+        private Filter(FilterBuilder builder) {
+            this.patternFilter = builder.patternFilter;
+            this.predefinedFilter = builder.predefinedFilter;
+            this.enabling = builder.enabling;
+        }
+
+        public String getPatternFilter() {
+            return patternFilter;
+        }
+
+        public String getPredefinedFilter() {
+            return predefinedFilter;
+        }
+
+        public Boolean getEnabling() {
+            return enabling;
+        }
+
+    }
+
+    public static final class FilterBuilder {
+
+        private String patternFilter;
+        private String predefinedFilter;
+        private Boolean enabling;
+
+        public FilterBuilder patternFilter(String patternFilter) {
+            this.patternFilter = patternFilter;
+            return this;
+        }
+
+        public FilterBuilder predefinedFilter(String predefinedFilter) {
+            this.predefinedFilter = predefinedFilter;
+            return this;
+        }
+
+        public FilterBuilder enabling(Boolean enabling) {
+            this.enabling = enabling;
+            return this;
+        }
+
+        public Filter build() {
+            if ((patternFilter == null || patternFilter.isEmpty())
+                    && (predefinedFilter == null || predefinedFilter.isEmpty())) {
+                throw new IllegalArgumentException("pattern-filter or predefined-filter must not be null or empty");
+            }
+            return new Filter(this);
+        }
+
+    }
+
+    public static final class Property {
+
+        private final String key;
+        private final String value;
+
+        public Property(String key, String value) {
+            this.key = key;
```

Comment:
You are definitely right - I forgot to add null check here. However, after discussion with mchoma, I will change this Property inner class to Map in some another PR. For that reason I leave it without checkers now since Property class is subject of future change.